tests/languages/download_test.py

```diff
@@ -0,0 +1,54 @@
+from __future__ import annotations
+
+import _thread as thread
+import platform
+from http.server import BaseHTTPRequestHandler
+from http.server import HTTPServer
+from pathlib import Path
+
+from pre_commit.languages import download
+from testing.language_helpers import run_language
+
+
+class MyRequestHandler(BaseHTTPRequestHandler):
+    def do_GET(self):
+        self.send_response(200)
+        self.end_headers()
+        self.wfile.write(b'#!/bin/sh\necho hello\nexit 123')
+
+
+class server():
+    def __init__(self):
+        self.httpd = HTTPServer(('localhost', 5555), MyRequestHandler)
+
+    def start(self):
+        thread.start_new_thread(self.httpd.serve_forever, ())
+
+    def stop(self):
+        self.httpd.shutdown()
+
+
+def test_download_dependencies(tmp_path: Path) -> None:
+    http = server()
+    http.start()
+    ret = run_language(
+        tmp_path,
+        download,
+        'test.bat',
+        deps=(
+            """linux/amd64
+sha256-oRJkj6Cr8nWIivZ9d3W+rVZt/aSW1l9YtxSVh+GtIHM=
+http://127.0.0.1:5555
+test.bat""",
+            """windows/amd64
+sha256-oRJkj6Cr8nWIivZ9d3W+rVZt/aSW1l9YtxSVh+GtIHM=
+http://127.0.0.1:5555
+test.bat""",
+        ),
+    )
+    http.stop()
+    # We don't expect this to pass under Windows.
+    if platform.system() != 'Windows':
+        assert ret == (123, b'hello\n')
+    else:
+        assert True
```

Comment:
Turned out that I mistakenly thought this would work. I removed the ``if`` statement here, but I had to add a new ``if`` statement at ``MyRequestHandler``